fix: Duplicate Name Error Message on saving document

marination opened the pull request:
## **Issue:**
<details><summary>Duplication Error Message two times</summary>
  
![image](https://user-images.githubusercontent.com/25857446/86450301-ec62c100-bd36-11ea-8991-1aaeb9e197b6.png)

</details>

## **Fix:**
- Remove try catch in `savedocs` as it only caught and re-raises NameError that handles itself everywhere
- Message and error raised on `db_insert` with styling fixes
  ![Screenshot 2020-07-03 at 2 08 25 PM](https://user-images.githubusercontent.com/25857446/86450745-a22e0f80-bd37-11ea-8a34-87c9bc05ea8f.png)
- Every invocation of `frappe.NameError` elsewhere is within `frappe.throw` i.e. for other name errors a message is raised. So removing the message in `savedocs` does no harm as far as i know

marination committed:
fix: Duplicate Name Error Message on saving document

gavindsouza commented on frappe/desk/form/save.py:
```diff
@@ -21,8 +21,7 @@ def savedocs(doc, action):
 			try:
 				doc.save()
 			except frappe.NameError as e:
-				doctype, name, original_exception = e if isinstance(e, tuple) else (doc.doctype or "", doc.name or "", None)
-				frappe.msgprint(frappe._("{0} {1} already exists").format(doctype, name))
+				# Duplicate Error message already shown on db_insert
 				raise
```
there's no point in catching an exception if all we're doing is raising it again
```suggestion
			doc.save()
```

marination committed:
fix: Remove unnecessary try catch

- Try catch only caught name errors that handle themselves wherever they occur
- Catching a well handled exception to raise it again is pointless

gavindsouza commented:
@Mergifyio backport version-12-hotfix
